src/librustc_mir/build/mod.rs

```diff
@@ -383,6 +383,29 @@ fn construct_fn<'a, 'gcx, 'tcx, A>(hir: Cx<'a, 'gcx, 'tcx>,
     let source_info = builder.source_info(span);
     let call_site_s = (call_site_scope, source_info);
     unpack!(block = builder.in_scope(call_site_s, LintLevel::Inherited, block, |builder| {
+
+        // According to rustc_trans/callee we mark all foreign instances with nounwind,
+        // so let's make sure we don't unwind.
+        // Therefore generate an extra "Abort" landing pad.
+
+        // FIXME: Figure out why we can't use something like this instead:
+        // tcx.is_foreign_item(tcx.hir.local_def_id(fn_id));
```

Comment:
`is_foreign_item` checks for *foreign items*, aka
```Rust
extern "C" {
    fn foreign_item(); // we don't generate MIR for this
}
```

Rather than *extern fns*, aka
```Rust
extern "C" fn extern_fn() {
    // we *do* generate MIR for this
}
```